src/ptr_array.rs

```diff
@@ -0,0 +1,218 @@
+// Copyright 2019, The Gtk-rs Project Developers.
+// See the COPYRIGHT file at the top-level directory of this distribution.
+// Licensed under the MIT license, see the LICENSE file or <http://opensource.org/licenses/MIT>
+
+use glib_sys;
+use std::fmt;
+use translate::*;
+
+glib_wrapper! {
+    pub struct PtrArray(Shared<glib_sys::GPtrArray>);
+
+    match fn {
+        ref => |ptr| glib_sys::g_ptr_array_ref(ptr),
+        unref => |ptr| glib_sys::g_ptr_array_unref(ptr),
+        get_type => || glib_sys::g_ptr_array_get_type(),
+    }
+}
+
+impl<'a, T> ToGlibContainerFromSlice<'a, *mut glib_sys::GPtrArray> for T
+where
+    T: GlibPtrDefault + ToGlibPtr<'a, <T as GlibPtrDefault>::GlibType>,
+{
+    type Storage = (
+        Option<PtrArray>,
+        Vec<Stash<'a, <T as GlibPtrDefault>::GlibType, T>>,
+    );
+
+    #[inline]
+    fn to_glib_none_from_slice(t: &'a [T]) -> (*mut glib_sys::GPtrArray, Self::Storage) {
+        let (ptr, storage) = Self::to_glib_container_from_slice(t);
+
+        unsafe { (ptr, (Some(from_glib_full(ptr)), storage.1)) }
+    }
+
+    #[inline]
+    fn to_glib_container_from_slice(t: &'a [T]) -> (*mut glib_sys::GPtrArray, Self::Storage) {
+        let stash_vec: Vec<_> = t.iter().map(ToGlibPtr::to_glib_none).collect();
+        unsafe {
+            let arr = glib_sys::g_ptr_array_new_full(t.len() as _, None);
+
+            for stash in &stash_vec {
+                let ptr: *mut <T as GlibPtrDefault>::GlibType = Ptr::to(stash.0);
+                glib_sys::g_ptr_array_add(arr, ptr as *mut _);
+            }
+
+            (arr, (None, stash_vec))
+        }
+    }
+
+    #[inline]
+    fn to_glib_full_from_slice(_t: &[T]) -> *mut glib_sys::GPtrArray {
+        unimplemented!()
```

Comment:
This seems like the one that is actually useful. The others are all potential footguns

Reply:
This is now implemented together with the rest of the methods.